File: BookExamples/34_ViewModelDemo_LiveData/app/src/main/java/com/ebookfrenzy/viewmodeldemo/ui/main/MainViewModel.java
package com.ebookfrenzy.viewmodeldemo.ui.main;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class MainViewModel extends ViewModel {

    private static final Float usd_to_eu_rate = 0.74F;
    private String dollarText = "";

    //private Float result = 0F;
    private MutableLiveData<Float> result = new MutableLiveData<>();


    public void setAmount(String value) {
        this.dollarText = value;
        //result = Float.parseFloat(dollarText)*usd_to_eu_rate;
        result.setValue(Float.parseFloat(dollarText)*usd_to_eu_rate);

    }

    //public Float getResult()
    public MutableLiveData<Float> getResult(){
        return result;
    }

}

File: class_book_examples/chapter41_ViewModelDemo_LiveData/app/src/main/java/com/ebookfrenzy/viewmodeldemo/ui/main/MainViewModel.kt
package com.ebookfrenzy.viewmodeldemo.ui.main

import androidx.lifecycle.ViewModel
import androidx.lifecycle.MutableLiveData

class MainViewModel : ViewModel() {
    private val rate = 0.74f
    private var dollarText = ""
    private var result: MutableLiveData<Float> = MutableLiveData()

    fun setAmount(value: String) {
        this.dollarText = value
        result.setValue(value.toFloat() * rate)
    }

    fun getResult(): MutableLiveData<Float> {
        return result
    }
}